sdk/telemetry/test/conversion_test.go

```diff
@@ -0,0 +1,247 @@
+		DroppedLinks:  4,
+		Status: &telemetry.Status{
+			Message: "okay",
+			Code:    telemetry.StatusCodeOK,
+		},
+	}
+	pSpanA = func() ptrace.Span {
+		s := ptrace.NewSpan()
+		s.SetTraceID(pcommon.TraceID([16]byte{0x1}))
+		s.SetSpanID(pcommon.SpanID([8]byte{0x2}))
+
+		ts := s.TraceState()
+		ts.FromRaw("test=a")
+
+		s.SetParentSpanID(pcommon.SpanID([8]byte{0x1}))
+		// https://github.com/open-telemetry/opentelemetry-collector/issues/11267
+		// s.SetFlags(1)
+		s.SetName("span.a")
+		s.SetKind(ptrace.SpanKindClient)
+		s.SetStartTimestamp(pcommon.NewTimestampFromTime(y2k))
+		s.SetEndTimestamp(pcommon.NewTimestampFromTime(y2k.Add(time.Second)))
+		pAttrsA.CopyTo(s.Attributes())
+		s.SetDroppedAttributesCount(2)
+		prevent.CopyTo(s.Events().AppendEmpty())
+		s.SetDroppedEventsCount(3)
+		pLink.CopyTo(s.Links().AppendEmpty())
+		s.SetDroppedLinksCount(4)
+
+		stat := s.Status()
+		stat.SetMessage("okay")
+		stat.SetCode(ptrace.StatusCodeOk)
+
+		return s
+	}()
+	schema100 = "http://go.opentelemetry.io/schema/v1.0.0"
+
+	scope = &telemetry.Scope{
+		Name:         "go.opentelemetry.io/auto/sdk/telemetry/test",
+		Version:      "v0.0.1",
+		Attrs:        []telemetry.Attr{telemetry.String("department", "ops")},
+		DroppedAttrs: 1,
+	}
+	pScope = func() pcommon.InstrumentationScope {
+		s := pcommon.NewInstrumentationScope()
+		s.SetName("go.opentelemetry.io/auto/sdk/telemetry/test")
+		s.SetVersion("v0.0.1")
+		s.Attributes().PutStr("department", "ops")
+		s.SetDroppedAttributesCount(1)
+		return s
+	}()
+
+	scopeSpans = &telemetry.ScopeSpans{
+		Scope:     scope,
+		Spans:     []*telemetry.Span{spanA},
+		SchemaURL: schema100,
+	}
+	pScopeSpans = func() ptrace.ScopeSpans {
+		s := ptrace.NewScopeSpans()
+		pSpanA.CopyTo(s.Spans().AppendEmpty())
+		pScope.CopyTo(s.Scope())
+		s.SetSchemaUrl(schema100)
+		return s
+	}()
+
+	res = telemetry.Resource{
+		Attrs: []telemetry.Attr{
+			telemetry.String("host", "hal"),
+			telemetry.Int("id", 42),
+		},
+		DroppedAttrs: 100,
+	}
+	press = func() pcommon.Resource {
+		r := pcommon.NewResource()
+		r.Attributes().PutStr("host", "hal")
+		r.Attributes().PutInt("id", 42)
+		r.SetDroppedAttributesCount(100)
+		return r
+	}()
+
+	resSpans = &telemetry.ResourceSpans{
+		Resource:   res,
+		SchemaURL:  schema100,
+		ScopeSpans: []*telemetry.ScopeSpans{scopeSpans},
+	}
+	pResSpans = func() ptrace.ResourceSpans {
+		rs := ptrace.NewResourceSpans()
+		press.CopyTo(rs.Resource())
+		pScopeSpans.CopyTo(rs.ScopeSpans().AppendEmpty())
+		rs.SetSchemaUrl(schema100)
+		return rs
+	}()
+
+	traces = telemetry.Traces{
+		ResourceSpans: []*telemetry.ResourceSpans{
+			resSpans,
+		},
+	}
+	pTraces = func() ptrace.Traces {
+		traces := ptrace.NewTraces()
+		pResSpans.CopyTo(traces.ResourceSpans().AppendEmpty())
+		return traces
+	}()
+)
+
+func TestDecode(t *testing.T) {
+	var enc ptrace.JSONMarshaler
+	b, err := enc.MarshalTraces(pTraces)
+	require.NoError(t, err)
+
+	t.Log(string(b))
+
```

Comment:
nit: is this needed?

```suggestion
```

Reply:
When this fails, the decode error does not always provide a clear understanding of what failed to decode. This only will be logged with verbose testing or when the failure occurs. I've found it useful when debugging the tests so far and think leaving it in may help in the future. It can be removed if you really think it shouldn't be added.

Reply:
I think it is fine. I thought that could be reason. You can leave a comment that this log line is helpful when test fails.

Reply:
```suggestion
	t.Log(string(b)) // This helps when test fails to understand what being decoded.

```